front/src/applications/operationalStudies/hooks/useAutoUpdateProjection.ts

```diff
@@ -61,7 +59,19 @@ const useAutoUpdateProjection = (
       if (!currentTrainIdForProjection && isTrainSchedule(timetableItemId)) {
         dispatch(updateTrainIdUsedForProjection(timetableItemId));
       }
-      return;
+      // if there is already a projection and the projected train still exists, keep it
+      if (currentTrainIdForProjection && isSelectedTimetableItemIncluded) {
+        if (isTrainSchedule(timetableItemIds[0])) {
+          dispatch(updateSelectedTrainId(timetableItemIds[0]));
+        } else {
+          const occurrenceIdToSelect = formatPacedTrainIdToOccurrenceId({
+            pacedTrainId: timetableItemIds[0],
+            occurrenceIndex: 0,
+          });
+          dispatch(updateSelectedTrainId(occurrenceIdToSelect));
+        }
+        return;
+      }
```

Comment:
You don't need to add this piece of code, you only need to remove the test `isTrainSchedule(timetableItemId)` l59.
This function is used to select the new selectedTrain and the new train used for the projection, in case the previous train has been deleted.
At this point, you know that the selectedTrain has not been removed. You only need to check that the train used for the projection has not been removed too, otherwise you use the selected train for the projection.
```suggestion
      return;
```